Makefile

```diff
@@ -101,9 +101,24 @@ endif
 ifneq (, $(findstring adx, $(PROC)))
 	ADX = "y"
 	TFW_CFLAGS += -DADX=1
+else
+ifdef M
+	DIR = $(M)/
+endif
+	ADX_TEST_SRC = $(DIR)check/adx_test.c
+	ADX_TEST_BIN = $(DIR)check/adx_test.exe
+	ADX_BUILD_OUTPUT := $(shell gcc -madx $(ADX_TEST_SRC) -o $(ADX_TEST_BIN))
+	ADX_SUPPORTED := $(shell $(ADX_TEST_BIN) | if grep -q "ADX supported"; then echo -n "y"; fi)
```

Comment:
We already have `scripts/check_conf.pl` for this purpose, so I'd suggest to move the compilation into the script. We also can embed `adx_test.c` code right into the script and compile it in `/tmp` with a generated file name and remove it after the check.

The Makefile can run the `scripts/check_conf.pl` to make sure that configuration is good.